### PR TITLE
Better diagnostic messages

### PR DESCRIPTION
**Pull request by FabioLuporini**

### CODE REVIEW

File: devito/operator.py [hunk 1]
```diff
@@ -253,10 +253,28 @@ def _prepare_arguments(self, **kwargs):
...
         args = ReducerMap()
-        args.update([p._arg_values(**kwargs) for p in self.input if p.name in kwargs])
-        args.update([p._arg_values() for p in self.input if p.name not in args])
+        for p in overrides:
+            args.update(p._arg_values(**kwargs))
+            try:
+                args = ReducerMap(args.reduce_all())
+            except ValueError:
+                raise ValueError("Override `%s` is incompatible with overrides `%s`" %
+                                 (p, [i for i in overrides if i.name in args]))
```
Comment (mloubout) [inline devito/operator.py]: From my experience (may have changed) the % followed by line break fails to process the string expression

Comment (mloubout) [reply devito/operator.py]: But if test pass may have been fixed in python

File: tests/test_operator.py [hunk 2]
```diff
@@ -947,6 +947,43 @@ def test_function_dataobj(self, so, to, pad, expected):
...
+        a0 = Function(name='a', grid=grid0)
+        b0 = Function(name='b', grid=grid0)
+        a1 = Function(name='a', grid=grid1)
+        b1 = Function(name='b', grid=grid1)
+
+        op = Operator(Eq(a0, a0 + b0 + 1))
+        op.apply()
+
+        try:
+            op.apply(a=a1, b=b1)
```
Comment (mloubout) [inline tests/test_operator.py]: running it locally, this runs fine, doesn't end up in the except section

Comment (FabioLuporini) [reply tests/test_operator.py]: as expected, otherwise CI would be red ? not sure I follow?

Comment (mloubout) [reply tests/test_operator.py]: >Check it's hitting the right error msg

This is never actually checked as op.apply runs fine

Comment (FabioLuporini) [reply tests/test_operator.py]: ah, I see. Good spot. DO you know why? Otherwise I'll take a look

Comment (mloubout) [reply tests/test_operator.py]: Both have same grid so still fine, `op.apply(a=a1, b=b0)` should do the trick

Comment (FabioLuporini) [reply tests/test_operator.py]: ah yes! can you patch it?

Comment (FabioLuporini) [reply tests/test_operator.py]: or I will tmr morning no probs

**Comment by FabioLuporini** [comment]
thanks @mloubout , I fixed it. Will merge once CI is green again

**Pull request merged**